pkg/network/controllers/vm.go

```diff
@@ -106,6 +94,29 @@ func (v *VMController) Sync(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstanc
 	return vmCopy, nil
 }
 
+func (v *VMController) syncVMIInterfaces(
+	vm *v1.VirtualMachine,
+	vmi *v1.VirtualMachineInstance,
+	indexedStatusIfaces map[string]v1.VirtualMachineInstanceNetworkInterface,
+) error {
+	vmiCopy := vmi.DeepCopy()
+	hasOrdinalIfaces := namescheme.HasOrdinalSecondaryIfaces(vmi.Spec.Networks, vmi.Status.Interfaces)
+	updatedVmiSpec := applyDynamicIfaceRequestOnVMI(vm, vmiCopy, hasOrdinalIfaces)
+	vmiCopy.Spec = *updatedVmiSpec
+
+	ifaces, networks := clearDetachedInterfaces(vmiCopy.Spec.Domain.Devices.Interfaces, vmiCopy.Spec.Networks, indexedStatusIfaces)
+	vmiCopy.Spec.Domain.Devices.Interfaces = ifaces
+	vmiCopy.Spec.Networks = networks
+
+	if err := v.vmiInterfacesPatch(&vmiCopy.Spec, vmi); err != nil {
```

Comment:
Pushing down the stack the patching operation is making things worse in regards to "hiding" it.
Patching the VMI is a bug and expensive deal, we should try and expose it on the main flow, not push it down the stack and hide it in the low details.

Nevertheless, we could try and handle it as a follow up.